modules/openapi-generator/src/main/resources/python/model_test.mustache

```diff
@@ -26,7 +33,11 @@ class Test{{classname}}(unittest.TestCase):
         """Test {{classname}}"""
         # FIXME: construct object with mandatory attributes with example values
         # model = {{packageName}}.models.{{classFilename}}.{{classname}}()  # noqa: E501
-        pass
+        {{#vars}}{{#required}}{{^isPrimitiveType}}{{#isModel}}{{^isEnum}}
+        imported{{dataType}} = import_module('test_' + camelToSnake("{{dataType}}"))
+		{{/isEnum}}{{/isModel}}{{/isPrimitiveType}}{{/required}}{{/vars}}
+        return {{packageName}}.models.{{classFilename}}.{{classname}}({{#vars}}{{#required}}
+            {{name}} = {{#defaultValue}}{{defaultValue}}{{/defaultValue}}{{^defaultValue}}{{#isPrimitiveType}}{{#isNumber}}1337{{/isNumber}}{{#isLong}}1337{{/isLong}}{{#isInteger}}1337{{/isInteger}}{{#isDouble}}1337.0{{/isDouble}}{{#isFloat}}1337.0{{/isFloat}}{{#isBoolean}}True{{/isBoolean}}{{#isString}}{{#isEnum}}{{#_enum}}{{#-first}}"{{.}}"{{/-first}}{{/_enum}}{{/isEnum}}{{^isEnum}}"{{dataType}}"{{/isEnum}}{{/isString}}{{#isByteArray}}"{{dataType}}"{{/isByteArray}}{{#isDate}}datetime.date.today(){{/isDate}}{{#isDateTime}}datetime.datetime.now(){{/isDateTime}}{{#isFile}}{{{dataType}}}{{/isFile}}{{/isPrimitiveType}}{{^isPrimitiveType}}{{#isModel}}{{#isEnum}}[]{{/isEnum}}{{^isEnum}}imported{{dataType}}.Test{{dataType}}.test{{dataType}}(self){{/isEnum}}{{/isModel}}{{^isModel}}{{^isListContainer}}{{#isEnum}}[]{{/isEnum}}{{/isListContainer}}{{/isModel}}{{#isUuid}}{{dataType}}{{/isUuid}}{{#isEmail}}{{dataType}}{{/isEmail}}{{/isPrimitiveType}}{{#isContainer}}{{#isListContainer}}[]{{/isListContainer}}{{#isMapContainer}}dict(){{/isMapContainer}}{{/isContainer}}{{/defaultValue}}{{#hasMore}}, {{/hasMore}}{{/required}}{{/vars}})  # noqa: E501
```

Comment:
These parameters already have example values form the java generator code. Please do not hard code them here. Please use `{{{example}}}` like we do here:
https://github.com/spacether/openapi-generator/blob/allof_anyof_oneof_addition/modules/openapi-generator/src/main/resources/python/python-experimental/api_doc_example.mustache#L17

Reply:
What if the YAML has no examples?

Reply:
Anyway yes, I will use {{{example}}}  when it is defined, and report to hardcoded values when there is no example.

Reply:
The example values are set here if there are no examples in the yaml for endpoint properties:
https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientCodegen.java#L660

If they have no example value set the example value could either be added in:
- PythonClientCodegen.java
- the mustache file
I prefer the PythonClientCodegen.java because that is where we set the other parameter defaults and it make it easier to maintain. Those defaults could be set by adding the method:
```
    @Override
    public String toExampleValue(Schema schema) {
        if (schema.getExample() != null) {
            return schema.getExample().toString();
        }
        // your code here setting missing values
    }
```
If you still want to use the mustache file to set fallback example values instead of updating the PythonClientCodegen.java  that is okay too.

Reply:
I see pros and cons to using PythonClientCodegen.java instead of the test templates:
- pros: it is easier to read; you can access the example from other templates; you can even save the example into the model themselves if the example was `null`;
- cons: the solution is more invasive, since it impacts much more than the generated "test" files; it affects PythonClientExperimentalCodegen.java, which inherits from PythonClientCodegen.java.

My concrete issue is with models whose properties are models themselves (e.g.: `ModelA` has a property of type `ModelB`). I was planning to call `TestModelB().make_instance()` in the `make_instance` method of the test for `ModelA`, but if I put that in the example of 
`ModelA`, the example is usable only if the test package is available.

Is there something I am not seeing that makes changing the java code a better solution? Or should I work on the test templates only?

Reply:
You are running into problems where all the data that you need is not accessible in the Python files and the mustache templates. The two pros that I see to working in the Java generator are:
- you have full access to all of the data models params etc
- it keeps the mustache template code shorter and easier to maintain

I am not sure if I understand your suggested example. TestModelB().make_instance()
Is TestModelB the test class? It could just be a function to make the instance.
For example in test_cat.py we can have make_instance() to give us a cat. As you pointed out getting to that function or method when testing another model is hard.
What do you think of these options?
- Option1. in test_cat have the class CatExample whose init method makes a cat. In `__init__` in the test folder load in CatExample. All tests can load in the parent test module and will have access to all of the ModelNameExample classes. Then if a model uses cat as a param we can use cat = CatExample() 
- Option2: in the generator for every model store a mutiline string in its example fully defining an instance of the model. This involves a lot of recursion. Potential you can output a mustache template to a string variable.

Reply:
Done, I will take now care of the other comments.

Reply:
Can you resolve this conversation?